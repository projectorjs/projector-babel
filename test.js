// @flow
'use strict';

const babel = require('./');

test('.transform()', () => {
  return babel.transform({
    code: 'input(1, 2, 3)',
    options: {
      minified: true,
    },
  }).then(result => {
    expect(result).toMatchSnapshot();
  });
});
