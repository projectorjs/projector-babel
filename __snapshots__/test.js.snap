// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.transform() 1`] = `
Object {
  "ast": Node {
    "comments": Array [],
    "end": 14,
    "loc": SourceLocation {
      "end": Position {
        "column": 14,
        "line": 1,
      },
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
    "program": Node {
      "body": Array [
        Node {
          "end": 14,
          "expression": Node {
            "arguments": Array [
              Node {
                "end": 7,
                "extra": Object {
                  "raw": "1",
                  "rawValue": 1,
                },
                "loc": SourceLocation {
                  "end": Position {
                    "column": 7,
                    "line": 1,
                  },
                  "start": Position {
                    "column": 6,
                    "line": 1,
                  },
                },
                "start": 6,
                "type": "NumericLiteral",
                "value": 1,
              },
              Node {
                "end": 10,
                "extra": Object {
                  "raw": "2",
                  "rawValue": 2,
                },
                "loc": SourceLocation {
                  "end": Position {
                    "column": 10,
                    "line": 1,
                  },
                  "start": Position {
                    "column": 9,
                    "line": 1,
                  },
                },
                "start": 9,
                "type": "NumericLiteral",
                "value": 2,
              },
              Node {
                "end": 13,
                "extra": Object {
                  "raw": "3",
                  "rawValue": 3,
                },
                "loc": SourceLocation {
                  "end": Position {
                    "column": 13,
                    "line": 1,
                  },
                  "start": Position {
                    "column": 12,
                    "line": 1,
                  },
                },
                "start": 12,
                "type": "NumericLiteral",
                "value": 3,
              },
            ],
            "callee": Node {
              "end": 5,
              "loc": SourceLocation {
                "end": Position {
                  "column": 5,
                  "line": 1,
                },
                "identifierName": "input",
                "start": Position {
                  "column": 0,
                  "line": 1,
                },
              },
              "name": "input",
              "start": 0,
              "type": "Identifier",
            },
            "end": 14,
            "loc": SourceLocation {
              "end": Position {
                "column": 14,
                "line": 1,
              },
              "start": Position {
                "column": 0,
                "line": 1,
              },
            },
            "start": 0,
            "type": "CallExpression",
          },
          "loc": SourceLocation {
            "end": Position {
              "column": 14,
              "line": 1,
            },
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "start": 0,
          "type": "ExpressionStatement",
        },
      ],
      "directives": Array [],
      "end": 14,
      "loc": SourceLocation {
        "end": Position {
          "column": 14,
          "line": 1,
        },
        "start": Position {
          "column": 0,
          "line": 1,
        },
      },
      "sourceType": "module",
      "start": 0,
      "type": "Program",
    },
    "start": 0,
    "tokens": Array [
      Token {
        "end": 5,
        "loc": SourceLocation {
          "end": Position {
            "column": 5,
            "line": 1,
          },
          "start": Position {
            "column": 0,
            "line": 1,
          },
        },
        "start": 0,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "name",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": true,
          "updateContext": [Function],
        },
        "value": "input",
      },
      Token {
        "end": 6,
        "loc": SourceLocation {
          "end": Position {
            "column": 6,
            "line": 1,
          },
          "start": Position {
            "column": 5,
            "line": 1,
          },
        },
        "start": 5,
        "type": TokenType {
          "beforeExpr": true,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "(",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": true,
          "updateContext": [Function],
        },
        "value": undefined,
      },
      Token {
        "end": 7,
        "loc": SourceLocation {
          "end": Position {
            "column": 7,
            "line": 1,
          },
          "start": Position {
            "column": 6,
            "line": 1,
          },
        },
        "start": 6,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "num",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": true,
          "updateContext": null,
        },
        "value": 1,
      },
      Token {
        "end": 8,
        "loc": SourceLocation {
          "end": Position {
            "column": 8,
            "line": 1,
          },
          "start": Position {
            "column": 7,
            "line": 1,
          },
        },
        "start": 7,
        "type": TokenType {
          "beforeExpr": true,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": ",",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": false,
          "updateContext": null,
        },
        "value": undefined,
      },
      Token {
        "end": 10,
        "loc": SourceLocation {
          "end": Position {
            "column": 10,
            "line": 1,
          },
          "start": Position {
            "column": 9,
            "line": 1,
          },
        },
        "start": 9,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "num",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": true,
          "updateContext": null,
        },
        "value": 2,
      },
      Token {
        "end": 11,
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 1,
          },
          "start": Position {
            "column": 10,
            "line": 1,
          },
        },
        "start": 10,
        "type": TokenType {
          "beforeExpr": true,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": ",",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": false,
          "updateContext": null,
        },
        "value": undefined,
      },
      Token {
        "end": 13,
        "loc": SourceLocation {
          "end": Position {
            "column": 13,
            "line": 1,
          },
          "start": Position {
            "column": 12,
            "line": 1,
          },
        },
        "start": 12,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "num",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": true,
          "updateContext": null,
        },
        "value": 3,
      },
      Token {
        "end": 14,
        "loc": SourceLocation {
          "end": Position {
            "column": 14,
            "line": 1,
          },
          "start": Position {
            "column": 13,
            "line": 1,
          },
        },
        "start": 13,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": ")",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": false,
          "updateContext": [Function],
        },
        "value": undefined,
      },
      Token {
        "end": 14,
        "loc": SourceLocation {
          "end": Position {
            "column": 14,
            "line": 1,
          },
          "start": Position {
            "column": 14,
            "line": 1,
          },
        },
        "start": 14,
        "type": TokenType {
          "beforeExpr": false,
          "binop": null,
          "isAssign": false,
          "isLoop": false,
          "keyword": undefined,
          "label": "eof",
          "postfix": false,
          "prefix": false,
          "rightAssociative": false,
          "startsExpr": false,
          "updateContext": null,
        },
        "value": undefined,
      },
    ],
    "type": "File",
  },
  "code": "input(1,2,3);",
}
`;
